src/components/Tabs/TabList/TabList.jsx

```diff
@@ -42,68 +41,42 @@ const TabList = forwardRef(({ className, id, onTabChange, activeTabId, tabType,
   );
 
   const onTabClick = useCallback(
-    (tabId, tabCallbackFunc) => {
+    (value, tabId) => {
+      const tabCallbackFunc = children[tabId].props.onClick;
       if (disabledTabIds.has(tabId)) return;
       if (tabCallbackFunc) tabCallbackFunc(tabId);
       onTabSelect(tabId);
-      setFocusTab(-1);
     },
-    [onTabSelect, disabledTabIds, setFocusTab]
+    [children, disabledTabIds, onTabSelect]
   );
 
-  function onKeyDown(keyCode) {
-    let newFocusTab = focusTab;
-    if (keyCode === 37 || keyCode === 39) {
-      // left or right arrow
-      if (newFocusTab < 0) {
-        newFocusTab = activeTabState;
-      }
-    }
-
-    if (keyCode === 37 && newFocusTab > 0) {
-      // left arrow
-      setFocusTab(newFocusTab - 1);
-    } else if (keyCode === 39 && newFocusTab < children.length - 1) {
-      // right arrow
-      setFocusTab(newFocusTab + 1);
-    } else if (keyCode === 13 || keyCode === 32) {
-      // enter or space
-      onTabSelect(focusTab);
-    }
-  }
-
-  const { keyboardProps } = useKeyboard({
-    onKeyDown: e => {
-      onKeyDown(e.keyCode);
-    },
-    onKeyUp: _e => {}
-  });
-
-  const { focusWithinProps } = useFocusWithin({
-    onFocusWithin: () => {
-      setFocusTab(activeTabState);
-    },
-
-    onBlurWithin: () => {
-      setFocusTab(-1);
-    }
+  const getItemByIndex = useCallback(index => children[index], [children]);
+  const disabledIndexes = useMemo(() => Array.from(disabledTabIds), [disabledTabIds]);
```

Comment:
I'm probably missing something, but shouldn't `disabledTabIds` just be called `disabledTabIndexes`? what's the difference here between an id and an index?

Reply:
the ids are index but saving in a set object which not support by the hook.